crates/ruff_python_stdlib/src/builtins.rs

```diff
@@ -370,6 +370,29 @@ pub fn is_python_builtin(name: &str, minor_version: u8, is_notebook: bool) -> bo
     )
 }
 
+/// Return `Some(version)`, where `version` corresponds to the Python minor version
+/// in which the builtin was added, if `name` corresponds to a Python builtin that
+/// was added in a later Python version than the one which the user specified in
+/// their Ruff config. Otherwise, return `None`.
+///
+/// This is useful in some situations to give a better error message
+/// if the user didn't specify a target version at all (in which case we will default
+/// to Python 3.8).
+pub fn version_builtin_was_added(name: &str, minor_version: u8) -> Option<u8> {
+    if minor_version < 10 {
+        (PY310_PLUS_BUILTINS.contains(&name)
+            || PY311_PLUS_BUILTINS.contains(&name)
+            || PY313_PLUS_BUILTINS.contains(&name))
+        .then_some(10)
+    } else if minor_version < 11 {
```

Comment:
I don't fully understand why we always return minor version 10 even for builtins added in `3.11` or 3.12?

Reply:
no you are very correct here, thanks!